File: game-server/src/main/java/org/forfun/mmorpg/game/function/event/PlayerFuncOpenEvent.java
package org.forfun.mmorpg.game.function.event;

import jforgame.commons.eventbus.BaseEvent;
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt;

public class PlayerFuncOpenEvent implements BaseEvent {

    private PlayerEnt player;

    public PlayerFuncOpenEvent(PlayerEnt player) {
        this.player = player;
    }

    @Override
    public PlayerEnt getOwner() {
        return player;
    }

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/function/event/PlayerFuncOpenEvent.kt
package org.forfun.mmorpg.game.function.event

import jforgame.commons.eventbus.BaseEvent
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt

class PlayerFuncOpenEvent(private val player: PlayerEnt) : BaseEvent {

    override fun getOwner(): PlayerEnt = player
}
